### PR TITLE
[chore] remove usage of Clone method

### PR DESCRIPTION
**Pull request by codeboten**
Replacing in favour of CopyTo

### CODE REVIEW

**Comment by codeboten** [comment]
pinging @dmitryax for a review

**Review by bogdandrutu — APPROVED** [review]

File: exporter/awskinesisexporter/internal/batch/encode_marshaler.go [hunk 1]
```diff
@@ -121,14 +129,22 @@ func (bm *batchMarshaller) Metrics(md pmetric.Metrics) (*Batch, error) {
...
+			errs = multierr.Append(errs, consumererror.NewMetrics(err, metrics))
 		}
 	}
 
 	return bt, errs
 }
+
+func appendErr(errs error) {
+
+}
```
Comment (dmitryax) [inline exporter/awskinesisexporter/internal/batch/encode_marshaler.go:150]: added unintentionally?

Comment (codeboten) [reply exporter/awskinesisexporter/internal/batch/encode_marshaler.go:150]: *sigh* yes. thanks

**Review by mx-psi — APPROVED** [review]

File: exporter/awskinesisexporter/internal/batch/encode_marshaler.go [hunk 2]
```diff
@@ -89,12 +93,16 @@ func (bm *batchMarshaller) Traces(td ptrace.Traces) (*Batch, error) {
 			if errors.Is(err, ErrUnsupportedEncoding) {
 				return nil, err
 			}
-			errs = multierr.Append(errs, consumererror.NewTraces(err, export.Clone()))
+			traces := ptrace.NewTraces()
+			export.CopyTo(traces)
+			errs = multierr.Append(errs, consumererror.NewTraces(err, traces))
```
Comment (dmitryax) [inline exporter/awskinesisexporter/internal/batch/encode_marshaler.go:98]: `export` is already a copy. I don't believe we need to clone it again here

**Pull request merged**